src/ekam/rules/intercept.c

```diff
@@ -164,29 +164,69 @@ static FILE* ekam_call_stream;
 static FILE* ekam_return_stream;
 
 static char current_dir[PATH_MAX + 1];
+static bool ignore_interceptions = false;
```

Comment:
Would it maybe have been easier to change the `execve()` interception to call `unsetenv("LD_PRELOAD")` in the case that the specified binary is the symbolizer?

Reply:
This is a good idea. `execve` will have to get mildly more expensive since every invocation will need to `readlink` to check the path. That being said, I can't see how that would be in any way different from what the initialization routine would need to do anyway.

Reply:
Looking into it, the exec syscalls are WRAPPED. Should I just check if `syscall_name` in `remap_file` starts with `exec` & unset `LD_PRELOAD` then? Could be a bit buggy with multiple LD_PRELOAD to do that vs what I wrote but I guess that's not really an important scenario?

Reply:
I would suggest expanding the macro and customizing, like the other syscalls that are customized.

Reply:
I have been completely unsuccessful in utilizing this approach to properly unset `LD_PRELOAD` and I'm not really sure it's possible since it looks like LLVM bypasses libc's execve: https://android.googlesource.com/platform/external/compiler-rt/+/master/lib/sanitizer_common/sanitizer_linux.cc#356

Reply:
Ugh, that's annoying, but I guess understandable considering they probably don't want to call back into the code being sanitized.

OK in that case I can't think of a better approach here.